src/Bundle/spec/Controller/ResourceControllerSpec.php

```diff
@@ -187,6 +187,39 @@ function it_returns_a_response_for_html_view_of_a_single_resource(
         $this->showAction($request);
     }
 
+    function it_returns_event_response_if_exists_during_show(
+        MetadataInterface $metadata,
+        RequestConfigurationFactoryInterface $requestConfigurationFactory,
+        RequestConfiguration $configuration,
+        AuthorizationCheckerInterface $authorizationChecker,
+        RepositoryInterface $repository,
+        SingleResourceProviderInterface $singleResourceProvider,
+        EventDispatcherInterface $eventDispatcher,
+        ResourceControllerEvent $event,
+        ResourceInterface $resource,
+        Request $request,
+        Response $response
+    ): void {
+        $metadata->getApplicationName()->willReturn('sylius');
+        $metadata->getName()->willReturn('product');
+
+        $requestConfigurationFactory->create($metadata, $request)->willReturn($configuration);
+        $configuration->hasPermission()->willReturn(true);
+        $configuration->getPermission(ResourceActions::SHOW)->willReturn('sylius.product.show');
+
+        $authorizationChecker->isGranted($configuration, 'sylius.product.show')->willReturn(true);
+        $singleResourceProvider->get($configuration, $repository)->willReturn($resource);
+
+        $eventDispatcher->dispatch(ResourceActions::SHOW, $configuration, $resource)->willReturn($event);
+
+        $event->getResponse()->willReturn($response);
+
+        $configuration->isHtmlRequest()->shouldNotBeCalled();
+        $this->createRestView($configuration, $resource)->shouldNotBeCalled();
```

Comment:
Oh, ok, this is the problem 💃 `createRestView` is a protected function inside `ResourceController`, so we should rather check if `$this->viewHandler` is called or not 🖖

Reply:
I've allowed myself to fix it :) 🚀

Reply:
Thank you Mateusz you're the king !